src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,31 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String whiteSpace = " ";
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromLocal = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToLocal = LocalDate.parse(dateTo, formatter);
+        StringBuilder builder = new StringBuilder();
+        //int[] salaryAll = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            int salary = 0;
+            for (String dat : data) {
+                String[] datPart = dat.split(whiteSpace);
+                LocalDate dateData = LocalDate.parse(datPart[0], formatter);
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
                LocalDate dateData = LocalDate.parse(datPart[DATE_INDEX], formatter);
```